src/compiler/typescript_to_rust/generate_rust_canister.ts

```diff
@@ -6,192 +6,96 @@ import {
     GLOBAL_AZLE_RUST_DIR,
     GLOBAL_AZLE_TARGET_DIR
 } from '../utils';
-import { Err, ok, Ok, Result } from '../utils/result';
-import { AzleError, Plugin, RunOptions, Rust } from '../utils/types';
+import { Result } from '../../lib';
+import { JSCanisterConfig, Plugin, SpawnSyncError } from '../utils/types';
 import { isVerboseMode } from '../utils';
-import { red } from '../utils/colors';
 
+// TODO I think we should use the official Azle Result and match everywhere
+// TODO we should get rid of the custom ones created here
 export function generateRustCanister(
     fileNames: string[],
     plugins: Plugin[],
     canisterPath: string,
-    { rootPath }: RunOptions
-): Result<undefined, AzleError> {
+    canisterConfig: JSCanisterConfig
+): Result<null, SpawnSyncError> {
     const compilerInfo = {
         plugins,
         file_names: fileNames
     };
 
     const compilerInfoPath = join(
         canisterPath,
-        rootPath,
+        canisterConfig.root,
         'azle_generate',
         'compiler_info.json'
     );
 
+    // TODO why not just write the dfx.json file here as well?
     writeFileSync(compilerInfoPath, JSON.stringify(compilerInfo));
```

Comment:
Just had a thought. Just a mental viewpoint, not that we need to change the code. But it seems like we're writing a config file for azle_generate here. So maybe we call it `.azlegeneraterc`? Kinda makes sense to me. Then when you run azle_generate it always looks for that file so we don't have to pass a file name across the CLI.

This makes more sense to me than calling it `compiler_info.json` and having to specify it via the CLI each time.

Just something to think about.